src/modules/elixir.rs

```diff
@@ -0,0 +1,81 @@
+use ansi_term::Color;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::process::Command;
+
+use super::{Context, Module};
+
+const ELIXIR_VERSION_PATTERN: &str = "\
+Erlang/OTP (?P<otp>\\d+)[^\\n]+
+
+Elixir (?P<elixir>\\d[.\\d]+).*";
+
+lazy_static! {
+    static ref ELIXIR_VERSION_REGEX: Regex = Regex::new(ELIXIR_VERSION_PATTERN).unwrap();
+}
+
+/// Create a module with the current Elixir version
+///
+/// Will display the Rust version if any of the following criteria are met:
+///     - Current directory contains a `mix.exs` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_elixir_project = context.try_begin_scan()?.set_files(&["mix.exs"]).is_match();
+
+    if !is_elixir_project {
+        return None;
+    }
+
+    let (otp_version, elixir_version) = get_elixir_version()?;
+
+    let mut module = context.new_module("elixir");
+    let module_style = module
+        .config_value_style("style")
+        .unwrap_or_else(|| Color::Purple.bold());
+    module.set_style(module_style);
+
+    module.new_segment("symbol", "🔮 ");
+    module.new_segment(
+        "version",
+        &format!("{} (OTP {})", elixir_version, otp_version),
+    );
+
+    Some(module)
+}
+
+fn get_elixir_version() -> Option<(String, String)> {
+    let output = Command::new("elixir")
+        .arg("--version")
+        .output()
+        .ok()
+        .and_then(|output| String::from_utf8(output.stdout).ok())?;
+
+    parse_elixir_version(&output)
+}
+
+fn parse_elixir_version(version: &str) -> Option<(String, String)> {
+    let captures = ELIXIR_VERSION_REGEX.captures(version)?;
+
+    let otp_version = captures["otp"].to_owned();
+    let elixir_version = captures["elixir"].to_owned();
+
+    Some((otp_version, elixir_version))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_elixir_version() {
+        const OUTPUT: &str = "\
+Erlang/OTP 22 [erts-10.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
```

Comment:
Phew, I can see why you want to use regexes for this!

Reply:
Actually maybe I could just do some splits but it would make the code a bit ugly and the actual `regex` perfs make the parsing actually sufficiently fast i think and it's cleaner this way i think.